plumbing/object/commit.go

```diff
@@ -286,6 +304,24 @@ func (c *Commit) encode(o plumbing.EncodedObject, includeSig bool) (err error) {
 		return err
 	}
 
+	if c.MergeTag != "" {
+		if _, err = fmt.Fprint(w, "\n"+headermergetag+" "); err != nil {
+			return err
+		}
+
+		// Split tag information lines and re-write with a left padding and
+		// newline. Use join for this so it's clear that a newline should not be
+		// added after this section. The newline will be added either as part of
+		// the PGP signature or the commit message.
+		// FIXME: should this and PGP signature be moved into a generic
+		// multi-line field handler?
```

Comment:
We can go with this approach and then refactor it as part of https://github.com/go-git/go-git/pull/705.